website/blog/2023-03-30-relay-15.mdx

```diff
@@ -0,0 +1,153 @@
+ * @rootFragment myRootFragment
+ */
+```
+
+#### Define multiple resolvers per file
+
+Prior to this release we only allowed a single Relay Resolver per file and required the Relay Resolver function to be the default export. In Relay 15 you're now able to define multiple Relay Resolvers per file and use named exports. In order to use this feature you'll need to enable the `use_named_imports_for_relay_resolvers` feature flag in your Relay config.
+
+```js
+/**
+ * @RelayResolver User.favorite_page: Page
+ * @rootFragment favoritePageFragment
+ */
+function usersFavoritePage(){
+  ...
+}
+
+/**
+ * @RelayResolver User.best_friend: Friend
+ * @rootFragment bestFriendFragment
+ */
+function usersBestFriend(){
+  ...
+}
+
+module.exports = {
+  usersFavoritePage,
+  usersBestFriend
+}
+```
+
+### Support `@refetchable` on interfaces
+
+Previously it wasn't possible to add the `@refetchable` directive to fragments on server interface types. That support has now been added and will no longer error in the Relay compiler.
+
+```
+// schema.graphql
+
+interface RefetchableInterfaceFoo @fetchable(field_name: "id") {
+    id: ID!
+}
+
+extend type Query {
+  fetch__RefetchableInterfaceFoo(id: ID!): RefetchableInterfaceFoo
+}
+
+// fragment
+
+fragment RefetchableFragmentFoo on RefetchableInterfaceFoo
+  @refetchable(queryName: "RefetchableFragmentFooQuery") {
+  id
+}
+```
+
+### Persisted query improvements
+
+For users who are using URL-based persisted queries, you can now specify custom headers to send with the request to persist your queries. For example, this can be used to send auth headers to your query persistence URL endpoint.
+
+```js
+persistConfig: {
+  url: 'example.com/persist',
+  headers: {
+    Authorization: 'bearer TOKEN'
+  }
+}
+```
+
+And for users who are using file-based persisted queries a new feature flag, `compact_query_text` was added to remove all whitespace from persisted query output text. This can cause a substantial reduction in your persisted query text file (we've seen an example of reducing a single query text from 84kb to 31kb). This new feature flag can be enabled within your Relay config file.
+
+```js
+persistConfig: {
+  file: 'path/to/file.json',
+  algorithm: 'SHA256'
+},
+featureFlags: {
+  compact_query_text: true
+}
+```
+
+### Typesafe updates now support missing field handlers
+
+One of the few breaking changes in this release, the [missing field handlers](https://relay.dev/docs/guided-tour/reusing-cached-data/filling-in-missing-data) function signature has been changed. The `record` argument to the handler used to recieve a `Record` type (which is an untyped grabbag containing whatever the store has) but we now pass a `ReadOnlyRecordProxy` which is much more typesafe and provides a better interface. Furthermore, we also replaced the `field` argument of type `NormalizationLinkedField` with `CommonLinkedField`, which both a `ReaderLinkedField` and `NormalizationLinkedField` "implement".
```

Comment:
This might not cover all of the changes made in this area, please review and suggest alternatives.